pkg/controller/direct/bigqueryconnection/connection_controller.go

```diff
@@ -315,22 +292,14 @@ func (a *Adapter) Export(ctx context.Context) (*unstructured.Unstructured, error
 	obj := &krm.BigQueryConnectionConnection{}
 	mapCtx := &direct.MapContext{}
 	obj.Spec = direct.ValueOf(BigQueryConnectionConnectionSpec_FromProto(mapCtx, a.actual))
-	tokens := strings.Split(a.id.External, "connections/")
-	obj.Spec.ResourceID = direct.LazyPtr(tokens[len(tokens)-1])
-	if mapCtx.Err() != nil {
-		return nil, mapCtx.Err()
-	}
-	parent, err := a.id.Parent()
+
+	parent, id, err := krm.ParseConnectionExternal(a.id.String())
 	if err != nil {
-		return nil, fmt.Errorf("BigQueryConnectionConnection %s parent unset: %w", a.id.External, err)
-	}
-	if parent != "" {
-		tokens := strings.Split(parent, "/")
-		if len(tokens) == 4 && tokens[0] == "projects" && tokens[2] == "locations" {
-			obj.Spec.ProjectRef = &refs.ProjectRef{Name: tokens[1]}
-			obj.Spec.Location = tokens[3]
-		}
+		return nil, err
 	}
+	obj.Spec.ResourceID = &id
```

Comment:
If we embed ID into Spec, I think these three lines will just be `obj.Spec.ID = id`